scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -228,6 +228,8 @@ private[spel] class Typer(
         case TypedNull =>
           invalidNodeResult(IllegalIndexingOperation)
         case TypedObjectWithValue(underlying, _) => typeIndexer(e, underlying)
+        case Unknown =>
```

Comment:
Can we remove the default case below? What case won't be exhausted?

Reply:
The default case catches non-map and non-list TypedClass types for which we want still apply previous behaviour with `dynamicPropertyAccessAllowed` feature flag.

As we discussed - in this step we add dynamic access to unknown fields only on nodes typed to `Unknown` and only via indexer operator.

Reply:
So let's replace it with explicit pattern to make visible what is handled in this case. `case otherClass: TypedClass => ` will exhaust this pattern matching?

Reply:
fixed